services/repository/transfer.go

```diff
@@ -360,15 +365,20 @@ func ChangeRepositoryName(ctx context.Context, doer *user_model.User, repo *repo
 	oldRepoName := repo.Name
 
 	// Change repository directory name. We must lock the local copy of the
-	// repo so that we can atomically rename the repo path and updates the
+	// repo so that we can automatically rename the repo path and updates the
 	// local copy's origin accordingly.
 
-	repoWorkingPool.CheckIn(fmt.Sprint(repo.ID))
+	ctx, releaser, err := globallock.Lock(ctx, getRepoWorkingLockKey(repo.ID))
+	if err != nil {
+		log.Error("lock.Lock(): %v", err)
+		return fmt.Errorf("lock.Lock: %w", err)
+	}
+
 	if err := changeRepositoryName(ctx, repo, newRepoName); err != nil {
-		repoWorkingPool.CheckOut(fmt.Sprint(repo.ID))
+		releaser()
 		return err
 	}
-	repoWorkingPool.CheckOut(fmt.Sprint(repo.ID))
+	releaser()
```

Comment:
Ditto.

Reply:
1e1d6e9 For some places, I use a different context name `lockCtx` and use that only for lock-necessary resources. Please review again.

Reply:
Why do you follow the bad example but overlook the good one?

Reply:
That's why I said such lock ctx is an over-design, "it will be abused frequently in the future".

If I was the architect, I will do everything to avoid people making low-level mistakes.

Reply:
As a comparison 8e09e01ef6a752b85f3686a7337862e038627e69 will follow the original design. I prefer the previous one because the two contexts make me feel more clearer and use them more carefully, but I don't against the second one.

Reply:
Wrong again.

Reply:
I see. You convinced me, since even Lunny can misuse it.